### PR TITLE
Fix PersistenceException in production

### LINKED ISSUE #301: [6.2.0-RC6] PersistenceException: models.XXX is NOT an Entity Bean registered with this server?
Running in dev mode: OK
Running in production mode: Exception
```
play.api.UnexpectedException: Unexpected exception[PersistenceException: models.User is NOT an Entity Bean registered with this server?]
```
Tested in Docker via `docker:publishLocal`
`application.conf`:
```
ebean.default = ["models.*"]
```

@mkurz you could break models configuration in this commit:
https://github.com/playframework/play-ebean/commit/692ae03c00485d1098ef570dffe69c73ba273f22
Original comment by @hedecai :
https://github.com/playframework/playframework/pull/10819#issuecomment-1126873794
@ruyconolly I think you should stay on 6.2.0-RC4 and Ebean 12.16.1 for now....

### PR DESCRIPTION
**Pull request by PromanSEW**
Fixes #301

Tested both in dev and prod mode by `publishLocal` snapshot and `docker:publishLocal` in Docker container (the same as production environment in Jelastic)
There was a crash before the fix, now all models are registered correctly, checked all CRUD operations and all models
I don't know how to provide such tests, you can check yourself @mkurz

Let's see changes in https://github.com/playframework/play-ebean/commit/692ae03c00485d1098ef570dffe69c73ba273f22 here:
https://github.com/playframework/play-ebean/blob/692ae03c00485d1098ef570dffe69c73ba273f22/play-ebean/src/main/java/play/db/ebean/DefaultEbeanConfig.java#L235
According Javadoc in `FilterBuilder.includePackage`, you can see implementation - `FilterBuilder.prefixPattern`, which does all work for you, so you should not change package pattern manually:
```java
private static String prefixPattern(String fqn) {
    if (!fqn.endsWith(".")) fqn += ".";
    return fqn.replace(".", "\\.").replace("$", "\\$") + ".*";
}
```

### CODE REVIEW

File: play-ebean/src/main/java/play/db/ebean/DefaultEbeanConfig.java [hunk 1]
```diff
@@ -232,8 +232,8 @@ private static Reflections getReflections(Environment env, String packageName) {
         private static ConfigurationBuilder getReflectionsConfiguration(String packageName, ClassLoader classLoader) {
             return new ConfigurationBuilder()
                     .addUrls(ClasspathHelper.forPackage(packageName, classLoader))
-                    .filterInputsBy(new FilterBuilder().includePackage((packageName + ".").replace(".","\\.") + ".*"))
-                    .setScanners(new TypeElementsScanner(), new TypeAnnotationsScanner(), new SubTypesScanner());
+                    .filterInputsBy(new FilterBuilder().includePackage(packageName))
+                    .setScanners(new TypeElementsScanner(), Scanners.TypesAnnotated, Scanners.SubTypes);
```
Comment (PromanSEW) [inline play-ebean/src/main/java/play/db/ebean/DefaultEbeanConfig.java:236]: `TypeAnnotationsScanner` and `SubTypesScanner` are deprecated and use `TypesAnnotated` and `SubTypes` in constructors respectively

**Comment by PromanSEW** [comment]
@mkurz please review

**Comment by mkurz** [comment]
Sorry for the delay, I will look at this today.

**Comment by PromanSEW** [comment]
@mkurz any updates?

**Review by mkurz — APPROVED** [review]
Thanks!

**Pull request merged**